File: app/src/main/java/com/kingz/view/LogininView.java
package com.kingz.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.EditText;
import android.widget.LinearLayout;
import com.kingz.customdemo.R;

public class LogininView extends LinearLayout {

    private EditText edit1, edit2;

    public LogininView(Context context) {
        super(context);
        loadView();
    }

    public LogininView(Context context, AttributeSet attrs) {
        super(context, attrs);
        loadView();
    }

    public LogininView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        loadView();
    }

    private void loadView(){
        setOrientation(VERTICAL);
        LayoutInflater.from(getContext()).inflate(R.layout.form_view, this);
        edit1 = (EditText) findViewById(R.id.edit1);
        edit2 = (EditText) findViewById(R.id.edit2);
    }

    @Override
    public void setFocusable(boolean focusable) {
        super.setFocusable(focusable);
        edit1.setFocusable(focusable);
        edit2.setFocusable(focusable);
    }
}

File: module-Login/src/main/java/com/zeke/module_login/view/LogininView.kt
package com.zeke.module_login.view

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.LinearLayout
import com.zeke.module_login.databinding.FormViewBinding

/**
 * 启动登录页面的用户账号登录视图
 */
class LogininView : LinearLayout {
    private lateinit var formViewbinding: FormViewBinding

    constructor(context: Context) : super(context) {
        loadView()
    }

    constructor(context: Context,
                attrs: AttributeSet) : super(context, attrs) {
        loadView()
    }

    constructor(context: Context,
                attrs: AttributeSet,
                defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        loadView()
    }

    private fun loadView() {
        orientation = VERTICAL
        formViewbinding = FormViewBinding.inflate(LayoutInflater.from(context), this);
    }

    fun getBinding():FormViewBinding = formViewbinding

    override fun setFocusable(focusable: Boolean) {
        super.setFocusable(focusable)
        formViewbinding.loginName.isFocusable = focusable
        formViewbinding.loginPwd.isFocusable = focusable
    }
}
